src/PhpSpreadsheet/Reader/Xlsx.php

```diff
@@ -2525,6 +2525,10 @@ private function readColumnsAndRowsAttributes(SimpleXMLElement $xmlSheet, Worksh
         if (isset($xmlSheet->cols) && !$this->readDataOnly) {
             foreach ($xmlSheet->cols->col as $col) {
                 for ($i = (int) ($col['min']); $i <= (int) ($col['max']); ++$i) {
+                    if ((int) ($col['max']) == 16384) {
```

Comment:
Shouldn't this be `> 16384`, otherwise we don't read attributes for the last possible column in the worksheet?

Reply:
This derives from my debugging attempt to solve the loading issue when I try to load a XLSX file created by MS Office.

The file only uses columns 1 to 3, but the xmlSheet of the file contains two sets of cols:
1. for column 1 to 3
2. for column 4 to 16384

Seems like MS Office creates an extra set to replesent all remaining unused columns.

In the old code, $columnsAttributes is touched for the second set with $col['min'] == 4
and then the loop is terminated since $col['max'] == 16384
Since column 4 is flagged, setColumnAttributes() is later called on column 4 and thus triggers column 4 to be cached as the highest column in \PhpOffice\PhpSpreadsheet\Worksheet

"if ((int) ($col['max']) == 16384)" is the old code's behavior and I have no idea why it's written as that. Looking for a fix to my issue with minimum change to code, I decided to simply terminate the loop early.

Reply:
Confirmed that the issue in \PhpOffice\PhpSpreadsheet\Reader\Xlsx::readColumnsAndRowsAttributes() is XLSX file dependent and should not be considered as issue #700, changes reverted.